content/token-swap.md

```diff
@@ -0,0 +1,848 @@
+const createTokenAccountPoolInstruction = Web3.SystemProgram.createAccount({
+    fromPubkey: wallet.publicKey,
+    newAccountPubkey: tokenAccountPool.publicKey,
+    space: token.ACCOUNT_SIZE,
+    lamports: rent,
+    programId: token.TOKEN_PROGRAM_ID,
+})
+const initializeTokenAccountPoolInstruction = token.createInitializeAccountInstruction(
+    tokenAccountPool.publicKey,
+    poolTokenMint,
+    wallet.publicKey
+)
+
+transaction.add(createTokenAccountPoolInstruction)
+transaction.add(initializeTokenAccountPoolInstruction)
+```
+
+### Pool Token Fee Account
+
+The pool token fee account is the token account that the fees for the token swaps are paid to. This account must be owned by a specific account defined in the swap program - that account has public key [HfoTxFR1Tm6kGmWgYWD6J7YHVy1UwqSULUGVLXkJqaKN](https://explorer.solana.com/address/HfoTxFR1Tm6kGmWgYWD6J7YHVy1UwqSULUGVLXkJqaKN?cluster=devnet).
+
+```tsx
+const feeOwner = new web3.PublicKey('HfoTxFR1Tm6kGmWgYWD6J7YHVy1UwqSULUGVLXkJqaKN')
+
+let tokenFeeAccountAddress = await token.getAssociatedTokenAddress(
+    poolTokenMint, // mint
+    feeOwner, // owner
+    true // allow owner off curve
+)
+
+const tokenFeeAccountInstruction = await token.createAssociatedTokenAccountInstruction(
+    wallet.publicKey, // payer
+    tokenFeeAccountAddress, // ata
+    feeOwner, // owner
+    poolTokenMint // mint
+)
+
+transaction.add(tokenFeeAccountInstruction)
+```
+
+### Create the swap pool
+
+With all of the prerequisite accounts created, you can create the swap pool initialization instruction using `TokenSwap.createInitSwapInstruction` from the `spl-token-swap` library.
+
+This function takes *a lot* of arguments. Let's talk through them.
+
+The first 7 arguments are the prerequisite token accounts we just discussed.
+
+After that comes the constant representing the Token Program ID followed by the constant representing the Token Swap Program ID.
+
+Next, there are 4 pairs of number arguments representing numerators and denominators for the trade fee, owner trade fee, owner withdraw fee, and host fee. The instruction uses the numerator and denominator for each to calculate the percentage of the fee. Lets explain each of the fees:
+1. **Trade fee** - fees that are retained by the swap pool token accounts during a trade and increase the redeemable value of LP-tokens. This fee rewards users for providing liquidity to the swap pool.
+2. **Owner trade fee** - fees that are retained by the swap pool token accounts during a trade, with the equivalent in LP-tokens minted to the owner of the program
+3. **Owner withdraw fee** - extra LP-tokens that are sent to the owner on every withdrawal
+4. **Host fee** - a proportion of the owner trade fees, sent to an extra host token account provided during the trade. This fee incentives external parties (such as a dex) to provide frontends for the swap pool and rewards them with a portion.
```

Comment:
@jamesrp13 I suggest we either spell out decentralized exchange, capitalize DEX, or both e.g.
```
... (such as a DEX)
... (such as a decentralized exchange)
... (such as a decentralized exchange, or "DEX")
```